### PR TITLE
ESLint and Prettier in Harmony

### LINKED ISSUE #85: JSLint and Prettier
We currently install Prettier as a dependency but it is only run in an ad-hoc manner. We should update this to run precommit/prepush, either in fix mode or to highlight issues.

At the same time it would be useful to add JSLint to highlight any silly mistakes or obvious code errors.

https://prettier.io/docs/en/eslint.html

### PR DESCRIPTION
**Pull request by i-like-robots**
Fixes #85 

I haven't enabled the prettier plugin because it needs to do quite a bit of work to make itself happy... one for another PR!

I've tried to fix the eslint errors where possible to do so.

### CODE REVIEW

File: athloi.js [hunk 1]
```diff
@@ -134,7 +134,6 @@ module.exports = ({tasks, prompt, addPrompt}) => ({
 		async run(options) {
 			const port = parseInt(url.parse(openUrls.storybook).port, 10);
 			if(await tcpPortUsed.check(port)) {
-				console.log();
```
Comment (quarterto) [inline athloi.js]: this was there deliberately. i like some whitespace in my cli output 🙂

File: components/x-interaction/src/InteractionClass.jsx [hunk 2]
```diff
@@ -16,13 +16,15 @@ export class InteractionClass extends Component {
 			// additional render will be scheduled.
 			Promise.resolve().then(() => {
 				this.setState({ [loading]: true }, () => {
+					/* eslint no-console:off */
 					console.log('didsetloading', this.state[loading]);
```
Comment (quarterto) [inline components/x-interaction/src/InteractionClass.jsx]: these shouldn't be here but they're removed in #59

File: tools/x-docs/src/components/search/index.js [hunk 3]
```diff
@@ -62,9 +65,9 @@ export default class Search extends Component {
 		</div>;
 	}
 
-	search = ev => {
+	search (ev) {
```
Comment (quarterto) [inline tools/x-docs/src/components/search/index.js]: this will need to be bound now

File: packages/x-logo/index.js [hunk 4]
```diff
@@ -72,25 +72,20 @@ const xPoints = ({x, y, width, height, thickness}) => {
 }
 
 export default class XLogo extends Component {
-	static defaultProps = {
```
Comment (quarterto) [inline packages/x-logo/index.js]: these missing default props are breaking the build

**Comment by quarterto** [comment]
was just about to approve but noticed the base branch is `matth/migration-guide-tweaks` which was merged, can you update to master plz

**Comment by i-like-robots** [comment]
Woah, good spot!

**Comment by quarterto** [comment]
i mean, it had all the _letters_ of `master` in it, just a few others as well

**Pull request merged**